### DISCUSSION TITLE
Why the collection is initialized even if I find the entity without populating?

### DISCUSSION THREAD

**Discussion by suerta-git** [mikro-orm/mikro-orm#2715] [Q&A] [answered]
Hi there,

I am confused by the result of my test:

```ts
it('should cascading remove', async () => {
  // Arrange
  const room = await roomRepository.findOneOrFail({
    roomId,
  })
  console.log(room.players.isInitialized())  // true

  // ...
})
```

Here the `players` is a one-to-many collection. I expect it is not initialized because I didn't populate it. But the result of `isInitialized()` is true.

Is this because of some caching mechanism? (The room with same roomId has just been stored in the setup phase)

Can I disable it to make it to be more predictable?

**Comment by B4nan** [comment, marked as answer]
> I didn't populate it. 

Not in the code you shared, but you did load it at some point before that probably. 

Given you are in tests, I would expect you forgot to clear the EM and keep working with the same context. Each test should have its own clean identity map.

**Reply by suerta-git** [reply]
> Not in the code you shared, but you did load it at some point before that probably.

Yes, before that I have these lines for setup:

```js
beforeEach(async () => {
  const newPlayer = new Player(connectionId)
  const newRoom = new Room(roomId)
  newRoom.players.add(newPlayer)
  await roomRepository.persistAndFlush(newRoom) // cascading persist players
})
```

And I guess after persisting EM cached `newRoom` so that I get the entire room entity instead of entity with uninitialized references, am I right?

**Update**:

Adding `orm.em.clear()` to clear caches works well.
